app/controllers/graders_controller.rb

```diff
@@ -87,6 +87,22 @@ def global_actions
       end
     end
 
+    if %w[assign random_assign].include? params[:global_actions]
+      inactive_graders_hash =
+        current_course.tas.joins(:user).where(roles: { hidden: true }).pluck(:id, :user_name)
+                      .map { |x| [x[0], x[1]] }.to_h
+
+      inactive_graders_for_flash = inactive_graders_hash.select { |k| grader_ids.include? k.to_s }
```

Comment:
You're mixing different types here. `grader_ids` should always be a list of integers, not strings. Make sure this is true for all test cases that you've added, as well as manually in the UI.